### DISCUSSION TITLE
NextAuth failed on Cloudflare pages

### DISCUSSION THREAD

**Discussion by Mrezagolbaba** [nextauthjs/next-auth#8547] [Help] [answered]
### Question 💬

I utilized NextAuth in my Next.js app but encountered an issue with the NextAuth route when publishing on Cloudflare Pages. Can you suggest a solution?
<img width="1247" alt="image" src="https://github.com/nextauthjs/next-auth/assets/9501814/2dd68b71-f6c5-4d3c-b241-0f868c06491d">

### How to reproduce ☕️

publish Nextjs app with the NextAuth on the Cloudflare page

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by IndigoNakamoto** [comment]
Similar issue. Has anyone been able to resolve it?

**Reply by Clem-** [reply, marked as answer]
You need to add the `export const runtime = 'edge';` to every routes/pages as clearly explained in the error message.

**Comment by parthmern** [comment]
that is giving error on whole site 
not just single page
